Source/Project64/N64 System/Mips/Pif Ram.cpp

```diff
@@ -297,19 +297,16 @@ void CPifRam::SI_DMA_READ()
 			{
 				continue;
 			}
-			RDRAM[RdramPos ^3] = m_PifRam[count];
+			RDRAM[BES(RdramPos)] = m_PifRam[count];
 		}
 	}
 	else
 	{
-		for (size_t i = 0; i < 64; i += 4)
-		{
-			unsigned __int32 pif_ram_dword;
-			std::memcpy(&pif_ram_dword, &PifRamPos[i], sizeof(unsigned __int32));
-
-			pif_ram_dword = swap32by8(pif_ram_dword);
+		register size_t i;

```

Comment:
Yes my method does theoretically expose the variable to unintentional modification.

However for some functions it also can make maintaining the loop variable type easier.  If it was `int i` or `size_t i` each time, you could have to weigh your security argument against one of maintainability of centralizing the type declaration instead of repeating it within each/multiple `for` loops.  Repeating the same exact text/keywords over and over also can make debugging more tedious or risk drawbacks (probably not security ones though, of the form you're citing).

>Right, however I'm quite sure parsing methods have gotten better since prior to 1999.

OK, but I doubt that parsing methods "getting better" implies which style parses faster.

>This is something I'll need a source on that supports the claim (preferably a modern one).

It's not something I suspect directly off the words of others.

I'm simply observing the logic that normal local declarations are more consistent.
```c
// Would you prefer:
int i, j;
int x = 0, y = 0, z = 0;
for (i = 0; i < LIMIT; i++)
    for (j = 0; j < LIMIT; j++)
        { /* statements depending on x, y and z */ }

// or should it be as you say, to maximize the locality of relevant scope with:
```c
for (int i = 0; i < LIMIT; i++)
    for (int j = 0, x = 0, y = 0, z = 0; i < LIMIT; i++)
```
Who will say?  Maybe the latter can compile faster, but I have no intention of writing it.

>Aside from it being deprecated?

Although, this reasoning is rather recursive.

We can't have something "deprecated" without having a reason for doing it.  You can't cite deprecation as the reason for avoiding something because that's recursive reasoning; there has to be a reason *why* it was deprecated.

So what exactly does "deprecated" mean in this case?  Human nature permits anything to be deprecated no matter how useless or even how use**ful** it may have been.